sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java

```diff
@@ -954,26 +962,66 @@ public void onPersonChanged(PersonDto existingPerson, Person newPerson) {
-						if (personCase.getOutcome() == CaseOutcome.NO_OUTCOME) {
-							CaseDataDto existingCase = CaseFacadeEjbLocal.toDto(personCase);
-							personCase.setOutcome(CaseOutcome.DECEASED);
-							personCase.setOutcomeDate(newPerson.getDeathDate() != null ? newPerson.getDeathDate() : new Date());
-							// Attention: this may lead to infinite recursion when not properly implemented
-							caseFacade.onCaseChanged(existingCase, personCase);
-						}
-					} else if (personCase.getOutcome() == CaseOutcome.DECEASED) {
+				Case personCase = personCases.isEmpty() ? null : personCases.get(0);
+				if (newPerson.getPresentCondition().isDeceased()
+					&& newPerson.getDeathDate() != null
+					&& newPerson.getCauseOfDeath() == CauseOfDeath.EPIDEMIC_DISEASE
+					&& newPerson.getCauseOfDeathDisease() != null) {
+
+					// update the latest associated case
+					if (personCase != null
+						&& personCase.getOutcome() != CaseOutcome.DECEASED
+						&& (personCase.getReportDate().before(DateHelper.addDays(newPerson.getDeathDate(), 30))
+							&& personCase.getReportDate().after(DateHelper.subtractDays(newPerson.getDeathDate(), 30)))) {
+						CaseDataDto existingCase = CaseFacadeEjbLocal.toDto(personCase);
+						personCase.setOutcome(CaseOutcome.DECEASED);
+						personCase.setOutcomeDate(newPerson.getDeathDate());
+						caseFacade.onCaseChanged(existingCase, personCase);
+					}
+				} else if (!newPerson.getPresentCondition().isDeceased()
+					&& (existingPerson.getPresentCondition() == PresentCondition.DEAD
+						|| existingPerson.getPresentCondition() == PresentCondition.BURIED)) {
+					// Person was put "back alive"
+					// make sure other values are set to null
+					newPerson.setCauseOfDeath(null);
+					newPerson.setCauseOfDeathDisease(null);
+					newPerson.setDeathPlaceDescription(null);
+					newPerson.setDeathPlaceType(null);
+					newPerson.setBurialDate(null);
+					newPerson.setCauseOfDeathDisease(null);
+					// update the latest associated case, if it was set to deceased && and if the case-disease was also the causeofdeath-disease
+					if (personCase != null
+						&& personCase.getOutcome() == CaseOutcome.DECEASED
+						&& personCase.getDisease() == existingPerson.getCauseOfDeathDisease()) {
 						CaseDataDto existingCase = CaseFacadeEjbLocal.toDto(personCase);
 						personCase.setOutcome(CaseOutcome.NO_OUTCOME);
 						personCase.setOutcomeDate(null);
-						// Attention: this may lead to infinite recursion when not properly implemented
 						caseFacade.onCaseChanged(existingCase, personCase);
 					}
 				}
+			} else if (newPerson.getPresentCondition() != null
+				&& newPerson.getPresentCondition().isDeceased()
+				&& newPerson.getDeathDate() != existingPerson.getDeathDate()
+				&& newPerson.getDeathDate() != null) {
+				// only Deathdate has changed
+				// update the latest associated case to the new deathdate, if causeOfDeath matches
+				Case personCase = personCases.isEmpty() ? null : personCases.get(0);
```

Comment:
Here as well (again not formulated well enough in the dev specifications) - always do this for the latest case within the threshold of the specified disease (i.e. the condition is not "latest case if it has this disease" but "latest case of this disease")